packages/nx/src/nx-cloud/utilities/url-shorten.ts

```diff
@@ -0,0 +1,111 @@
+import { logger } from '../../devkit-exports';
+import { getGithubSlugOrNull } from '../../utils/git-utils';
+
+export async function shortenedCloudUrl(
+  installationSource: string,
+  accessToken: string,
+  github?: boolean
+) {
+  const githubSlug = getGithubSlugOrNull();
+
+  const apiUrl = removeTrailingSlash(
+    process.env.NX_CLOUD_API || process.env.NRWL_API || `https://cloud.nx.app`
+  );
+
+  const installationSupportsGitHub = await getInstallationSupportsGitHub(
+    apiUrl
+  );
+
+  const usesGithub =
+    (githubSlug || github) &&
+    (apiUrl.includes('cloud.nx.app') ||
+      apiUrl.includes('eu.nx.app') ||
+      installationSupportsGitHub);
+
+  const source = getSource(installationSource);
+
+  try {
+    const response = await require('axios').post(
+      `${apiUrl}/nx-cloud/onboarding`,
+      {
+        type: usesGithub ? 'GITHUB' : 'MANUAL',
+        source,
+        accessToken: usesGithub ? null : accessToken,
+        selectedRepositoryName: githubSlug,
+      }
+    );
+
+    if (!response?.data || response.data.message) {
+      throw new Error(
+        response?.data?.message ?? 'Failed to shorten Nx Cloud URL'
+      );
+    }
+
+    return `${apiUrl}/connect/${response.data}`;
+  } catch (e) {
+    logger.warn(`Failed to shorten Nx Cloud URL.
```

Comment:
I think we should not the user the error here. The error on the App side should already be reported, and it's a bit jarring to see the error given that they aren't blocked from continuing the flow.

Maybe stick it into `logger.verbose` for us to debug if needed.

Reply:
Ok!